masonry/src/widgets/stack.rs

```diff
@@ -0,0 +1,646 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A widget that arranges its children in a one-dimensional array.
+
+use std::any::TypeId;
+
+use accesskit::{Node, Role};
+use tracing::{Span, trace_span};
+use vello::Scene;
+use vello::kurbo::{Affine, Line, Point, Size, Stroke};
+
+use crate::core::{
+    AccessCtx, Axis, BoxConstraints, ChildrenIds, LayoutCtx, NewWidget, NoAction, PaintCtx,
+    PropertiesMut, PropertiesRef, RegisterCtx, UpdateCtx, Widget, WidgetId, WidgetMut, WidgetPod,
+};
+use crate::properties::types::{CrossAxisAlignment, MainAxisAlignment};
+use crate::properties::{Background, BorderColor, BorderWidth, CornerRadius, Padding};
+use crate::theme::DEFAULT_GAP;
+use crate::util::{debug_panic, fill, include_screenshot, stroke};
+use crate::widgets::flex::get_spacing;
+
+/// A container with either horizontal or vertical layout.
+///
```

Comment:
Having a note here that references `Flex` for more complicated scenarios (and which those are) would be useful.